server/services/zwavejs-ui/api/zwaveJSUI.controller.js

```diff
@@ -0,0 +1,75 @@
+const asyncMiddleware = require('../../../api/middlewares/asyncMiddleware');
+
+module.exports = function ZwaveJSUIController(zwaveJSUIHandler) {
+  /**
+   * @api {post} /api/v1/service/zwavejs-ui/discover Start a new discovery of Z-Wave devices
+   * @apiName discover
+   * @apiGroup ZwaveJSUI
+   */
+  async function discover(req, res) {
+    await zwaveJSUIHandler.scan();
+    res.json({ success: true });
+  }
+  /**
+   * @api {post} /api/v1/service/zwavejs-ui/configuration Save configuration
+   * @apiName saveConfiguration
+   * @apiGroup ZwaveJSUI
+   */
+  async function saveConfiguration(req, res) {
+    await zwaveJSUIHandler.saveConfiguration(req.body);
+    res.json({ success: true });
+  }
+  /**
+   * @api {post} /api/v1/service/zwavejs-ui/connect Connect to MQTT broker
+   * @apiName connect
+   * @apiGroup ZwaveJSUI
+   */
+  async function connect(req, res) {
+    await zwaveJSUIHandler.connect();
+    res.json({ success: true });
+  }
+
+  /**
+   * @api {get} /api/v1/service/zwavejs-ui/node Connect to MQTT broker
+   * @apiName getNodes
+   * @apiGroup ZwaveJSUI
+   */
+  async function getNodes(req, res) {
```

Comment:
This seems to not be async ;)

Reply:
I prefer to keep controllers in asyncWrapper for error handling even if they are not async